### ISSUE TITLE
Zero pointer to struct fields, even if ZeroFields:false

### ISSUE THREAD

**Issue by skipor** [mitchellh/mapstructure#48]
Hello!
I have found unexpected  behaviour of Decode to struct with pointer to struct fields. 

``` go

import (
    "testing"

    "github.com/mitchellh/mapstructure"
    "github.com/stretchr/testify/assert"
    "github.com/stretchr/testify/require"
)

type Level1 struct {
    Rewrited string
    Default  string
}

type Level2 struct {
    StructPtr  *Level1
    OtherValue string
}

type M map[string]interface{}

func TestNestedStructDecode(t *testing.T) {
    const iniVal = "initialValue"
    const newVal = "writedValue"
    result := Level2{
        StructPtr:  &Level1{iniVal, iniVal},
        OtherValue: iniVal,
    }

    decoder, err := mapstructure.NewDecoder(
        &mapstructure.DecoderConfig{
            ZeroFields: false,
            Result:     &result,
        })
    require.NoError(t, err)

    err = decoder.Decode(M{
        "StructPtr": M{"Rewrited": newVal},
    })
    require.NoError(t, err)
    assert.Equal(t, newVal, result.StructPtr.Rewrited) //OK
    assert.Equal(t, iniVal, result.OtherValue)         //OK
    assert.Equal(t, iniVal, result.StructPtr.Default)  //Failed
}
```

Last assert fails with error:

```
     Error:          Not equal: "initialValue" (expected)
                                != "" (actual)
```

However, if make StructPtr field have type Layer1, but *Layer1, test passed and all is working as expected.
As for me, result should not changed, after changing type to pointer and reverse.
Thanks!

**Comment by gernoteger** [comment]
This also applies to slices of nested structures, same beheaviour as with pointers

**Comment by gernoteger** [comment]
I created PR #61 to fix this issue (including the slices case)

**Comment by mitchellh** [comment]
Fixed by #61!
